modules/asg_node_group/main.tf

```diff
@@ -106,6 +106,7 @@ resource "aws_autoscaling_group" "nodes" {
   name                      = "${local.name_prefix}-${each.key}"
   min_size                  = local.min_size
   max_size                  = local.max_size
+  desired_capacity          = var.desired_capacity
```

Comment:
I don't think that we want to set desired_capacity.

cluster_autoscaler is responsible for setting this.

If cluster_autoscaler increases this value then we run terraform we could cause unexpected scale in!